go/controller/autoscaler/autoscaler.go

```diff
@@ -31,7 +31,7 @@ type Cluster interface {
 }
 
 type job struct {
-	Config      *paddlejob.TrainingJob
+	Config      paddlejob.TrainingJob
```

Comment:
`paddlejob.TrainingJob` can be a large struct. We don't want to copy around this struct.

Reply:
```go
var j api.TrainingJob
fmt.Println(unsafe.Sizeof(j)) // 472
```
With 1MB of memory, we can have around 2200 `api.TrainingJob`, maybe it's not that big?

Here is a related guide: [pass values](https://github.com/golang/go/wiki/CodeReviewComments#pass-values), a `string` mentioned in the link is 16 bytes, so I think this is not a strong case for "pass values". Anyway, something to consider.